### PR TITLE
Adding doc for release process v2

### PR DESCRIPTION
**Pull request by rachitchauhan43**
Signed-off-by: rachitchauhan43 <rachitchauhan43@gmail.com>

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Type of changes**
Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**Feature/Issue validation/testing**:

Please describe the tests that you ran to verify your changes and relevent result summary. Provide instructions so it can be reproduced.
Please also list any relevant details for your test configuration.

- [ ] Test A
- [ ] Test B

- Logs

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Checklist**:

- [ ] Have you added unit/e2e tests that prove your fix is effective or that this feature works?
- [ ] Has code been commented, particularly in hard-to-understand areas?
- [ ] Have you made corresponding changes to the documentation?

**Release note**:

```release-note

```

### CODE REVIEW

File: release/RELEASE_PROCESS_v2.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+4. After it is merged,
+    1. Create a release branch of the form release-X.Y.Z from the master // TODO add git commands
+    2. Create a release candidate tag X.Y.Z-rc0 from that branch and do git push for both the branch and tag // TODO do mention only KServe owners can do it
+    3. from that tag create a release-candidate (basically a pre-release) on github
+    4. With this you are done with the creation of RC0 for upcoming release
+5. Announce in the community about the availability of release-candidate so that community can start consuming and testing. And ask them to report bugs as soon as possible.
+6. After feature freeze date, now only bug fixes will be merged into the release branch.
+
+### 1 week after feature freeze:
+After feature freeze,we will be merging only gug fixes into the release branch and creating another release candidate (RC1).
```
Comment (yuzisun) [inline release/RELEASE_PROCESS_v2.md]:
```suggestion
After feature freeze, we will be merging only bug fixes into the release branch and creating another release candidate (RC1).
```

**Comment by yuzisun** [comment]
@rachitchauhan43 Are we ready to get this in?

**Comment by rachitchauhan43** [comment]
@yuzisun : Apologies for the delay. Will close it this week.

**Comment by yuzisun** [comment]
> @yuzisun : Apologies for the delay. Will close it this week.

@rachitchauhan43 Is there any more changes need to be made on this PR?

**Comment by yuzisun** [comment]
> > @yuzisun : Apologies for the delay. Will close it this week.
> 
> @rachitchauhan43 Is there any more changes need to be made on this PR?

@rachitchauhan43 Let's just get this in and improve with the 0.11 release?

**Comment by rachitchauhan43** [comment]
> > > @yuzisun : Apologies for the delay. Will close it this week.
> > 
> > 
> > @rachitchauhan43 Is there any more changes need to be made on this PR?
> 
> @rachitchauhan43 Let's just get this in and improve with the 0.11 release?

Hey @yuzisun , I am done with the changes. Please review

**Comment by yuzisun** [comment]
Thanks @rachitchauhan43 !!

/lgtm
/approve

**Pull request merged**